zulipterminal/ui_tools/boxes.py

```diff
@@ -751,10 +751,23 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                     email=self.recipients_emails
                 )
             elif self.message['type'] == 'stream':
-                self.model.controller.view.write_box.stream_box_view(
-                    caption=self.message['display_recipient'],
-                    title=self.message['subject']
-                )
+                # Check if user is allowed to post
+                if (self.model.stream_dict[self.message.get('stream_id')].
+                        get('is_announcement_only', None) == 1
+                    or (self.model.initial_data.get('is_admin', None)
+                    or self.model.initial_data.get('is_owner', None)
+                    and self.model.stream_dict[self.message.get('stream_id')].
+                        get('stream_post_policy', None) == 2)):
```

Comment:
`is_announcement_only` is still present in what will soon be `3.0`, so we could stick with just it for now to begin with, since it works across the versions we plan to support.

If we support the new versions, we should likely have this logic in the model, in any case.

We'd ideally want tests against the two different server versions (or feature levels). While we added some background for this in #653, we don't have good examples of using this in a feature like this yet, so you're welcome to give it a go but you may wish to discuss on the stream about this first.